### ISSUE TITLE
feat: Ignore warning

### ISSUE THREAD

**Issue by gaetbout** [starkware-libs/cairo#6736]
# Feature Request

**Describe the Feature Request**

In some situation I'd like to be able to ignore specific warnings.

**Describe Preferred Solution**

Some sort of `#![allow(dead_code)]` or `#![allow(colliding_path)]` when dealing with component having the same storage slot.

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes
- [ ] No

**Comment by orizi** [comment]
you want to specifically allow dual access to the same storage area?

**Comment by gaetbout** [comment]
Yes, we need it, for example when upgrading to have access to the slots, while extracting the code in a component

**Comment by orizi** [comment]
I'm thinking of something like #[legacy] to mark a field we shouldn't check

**Comment by gaetbout** [comment]
That would solve this specific issue. 
But some other users might want to be able to ignore other warnings :/

**Comment by orizi** [comment]
some are already ignored using some `allow` or other - i'm not certain at all we want to ignore any specific warning in context.

**Comment by gaetbout** [comment]
Then I see the need for a way to indicate that I don’t want a warning about conflicting storage keys.
